File: angular/footballersApp/src/app/shared/models/footballer.ts
export class Footballer {
    name: string;
    fullName: string;
    position: string;
    dateOfBirth: string;
    placeOfBirth: string;
    nationality: string;
    imgUrl: string;
    footballerType: string;

    constructor() {}

    getName(): string {
        return this.name;
    }

    getFullName(): string {
        return this.fullName;
    }

    getPosition(): string {
        return this.position;
    }

    getDateOfBirth(): string {
        return this.dateOfBirth;
    }

    getPlaceOfBirth(): string {
        return this.placeOfBirth;
    }

    getNationality(): string {
        return this.nationality;
    }

    getImgUrl() { return this.imgUrl; }

    getFootballeType() { return this.footballerType; }

    setName(name: string) {
        this.name = name;
    }

    setFullName(fullName: string) {
        this.fullName = fullName;
    }

    setPosition(position: string) {
        this.position = position;
    }

    setDateOfBirth(dateOfBirth: string) {
        this.dateOfBirth = dateOfBirth;
    }

    setPlaceOfBirth(placeOfBirth: string) {
        this.placeOfBirth = placeOfBirth;
    }
    setNationality(nationality: string) {
        this.nationality = nationality;
    }

    setImgUrl(imgUrl: string) { this.imgUrl = imgUrl; }


    setFootballerType(footballerType: string) {
        this.footballerType = footballerType;
    }


}

File: spring/demo/src/main/java/com/example/footballers/models/Footballer.java
package com.example.footballers.models;


public class Footballer {


    public String name;
    public String fullName;
    public String position;
    public String dateOfBirth;
    public String placeOfBirth;
    public String nationality;
    public String imgUrl;
    public String footballerType; // Footballer type can be a starter, sub, or a reserve

    public String getName() {
        return name;
    }

    public String getPosition() {
        return position;
    }

    public String getDateOfBirth() {
        return dateOfBirth;
    }

    public String getNationality(){
        return nationality;
    }

    public String getFullName() { return this.fullName; }

    public String getPlaceOfBirth() { return this.placeOfBirth; }

    public String getImgUrl(){  return this.imgUrl; }

    public String getFootballerType() { return this.footballerType; }

    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public void setName(String name) { this.name = name; }

    public void setDateOfBirth(String dob) { this.dateOfBirth = dob; }

    public void setPosition(String position) { this.position = position; }

    public void setFullName(String fullName) { this.fullName = fullName; }

    public void setPlaceOfBirth(String placeOfBirth) { this.placeOfBirth = placeOfBirth; }

    public void setImgUrl(String imgUrl) { this.imgUrl = imgUrl; }

    public void setFootballerType(String footballerType) { this.footballerType = footballerType; }

}
